### DISCUSSION TITLE
Trying to use the Gotek to flash corrupted BIOS.

### DISCUSSION THREAD

**Discussion by CWEastwood** [keirf/flashfloppy#633] [Q&A] [answered]
Hello,

I recently purchase a Gotek Floppy emulator. Model per instruction sheet is SFR1M44-U100 code *: U00 142. I flashed it with FlashFloppy and I'm getting the F-F and when I plug in USB stick which I think has a valid image on it I get 0- on the 7 segment display. I **did not** load the usb with the ff.cfg file or img.cfg file (setup page says this is optional?). Jumper is set to S0 and JC (JC is not labeled on the PCB but is detailed in the instructions sheet.) The motherboard would not recognize the Gotek when set to S1.

What I'm trying to do is flash a corrupted BIOS. The BIOS is an Award BIOS currently in BootBlocker 1.0 and wants a floppy to reflash. I do not have that floppy disk to make an image. The image must be bootable, contain the bios .bin file, flash utility .exe file and a batch file. I have all of those files. When I place the files in  winimage and export a .img or .ima file the bios still requests a disk. 

I'm having trouble figuring out where my problem is. I figure I either don't have the Gotek jumpered properly, Flash Floppy is not configured properly, the image file I'm making is not correct in one or more ways OR Gotek/Flash Floppy is not useful to reflash a bios. 

It's hard to know where to start so I'm hoping someone might have some insight to share. Or preferably has tried doing this exact this in the past.

If I had to guess, I think I'm not creating a proper **bootable** floppy image but the 4 or so programs I've downloaded don't seem to have that functionality or I can't find it or I'm too illiterate to understand how to.  I will admit I am not very clever when it comes to IT. I have no background in programming,  or micro electronics.  

The USB I'm using is really an adapter for a 32gb SD card formatted in FAT32.

I know this forum is for issues with Gotek or FlashFloppy but I want to be thorough and also post this instructions I've been following to create the image file:

Thanks for reading,
CW Eastwood

RECOVERING A CORRUPT AWARD BOOTBLOCK BIOS

With AWARD BIOS the process is similar but still a bit different. To recover an AWARD BIOS you
will need to create a floppy diskette with a working BIOS file in .BIN format, an AWARD flash
utility and an AUTOEXEC.BAT file. AWARD BIOS will not automatically restore the BIOS
information to the SYSTEM BLOCK for this reason you will need to add the commands necessary
to flash the BIOS to the AUTOEXEC.BAT file. The system will run the AUTOEXE.BAT file at
startup, which will in turn flash the BIOS. This is fairly easy. Here are the steps you need to take.

1) Create a bootable floppy disk. If you use Windows 95, 98 or DOS, this can be done by
going to the DOS prompt and typing the command. (Be sure to insert a disk first)
FORMAT A: /S
The /S switch specifies that the system files should be copied to the diskette to make it
bootable. If you are using Windows ME, Windows NT or Windows 2000 then you can not
use the /S switch with the FORMAT command. If you are unable to create a bootable
diskette because of the operating system you are using then you may download a
bootable diskette image from the BootDisk_dot_com* web site at (http://www.bootdisk_dot_com)*

2) Copy a working BIOS ROM (.BIN) file for your motherboard and an AWARD flash utility
to the floppy disk. You can usually download these files from the motherboard
manufacturer. If you need help finding these files call the motherboard manufacturer for
technical support.

3) Use a text editor or the ‘edit’ command at the DOS prompt to create a text file named
‘AUTOEXEC.BAT’ and save it on the floppy disk. Make sure that the file name is
AUTOEXEC and that the extension is .BAT (nothing else). If you create this file from
RECOVERING A CORRUPT AWARD BOOTBLOCK BIOS (Cont.)

4) within Windows using a program such as NOTEPAD, you may need to change the file
extension from .TXT to .BAT

5) With the text editor add the following commands to the AUTOEXEC.BAT file on the
floppy disk and then save the changes.

@ECHO OFF
@AWDFLASH BIOSFILE.BIN /py

NOTE: The ‘/py’ switch will cause the flash utility to automatically reprogram the BIOS
without any user intervention. Be sure to change the ‘AWDFLASH’ to the name
of the flash utility that you are using and change the ‘BIOSFILE.BIN’ to the name
of the BIOS ROM file you are using.

If you are not sure what the command line switch to automatically flash your
BIOS is, type the name of the flash utility with the ‘/?’ switch after it at a DOS
prompt and a list of the available switches will be displayed.

6) Insert the diskette you created into the floppy drive A: on the system that you need to
restore the BIOS on. Turn on the system. The computer should startup, read the floppy
disk, run the commands in the AUTOEXEC.BAT file and restore the BIOS.

At this point if everything went well then the BIOS should have been restored and the
computer should restart normally. If your system still did not restart normally then go to
the TROUBLESHOOTING PROBLEMS section below.

TROUBLESHOOTING PROBLEMS

PROBLEM : While trying to recover a AWARD BIOS the system starts to read from the floppy disk
and then stops and nothing else happens. When the system is rebooted it still has
the same problem.

This most likely indicates that there is something wrong with the boot disk. Double
check that the disk is bootable, contains the AUTOEXEC.BAT file, the BIOS flash
utility and the BIOS ROM (.BIN) file. If all of the files are present and the disk is
bootable then double check the AUTOEXEC.BAT file. Make sure that the name is
correct and that all of the commands are correct.

If you can see on the screen what the system is doing then try typing the commands
in the AUTOEXEC.BAT file manually to see if there is an error message that is
displayed. If everything checks out then try using a different BIOS ROM (.BIN) file. If
the problem persists use a different flash utility.

*modified document to remove accidental hyperlinking

**Comment by hharte** [comment]
Hello,

You could try booting MS-DOS in a VM (like VmWare or VirtualBox) and use “format a: /s” to create a bootable floppy disk image.  Or, take a bootable MS-DOS disk image and remove some files from it using WinImage to free up space, and then add the files you need.  MS-DOS only needs MSDOS.SYS, IO.SYS and COMMAND.COM to boot, but the .SYS files have Hidden and System attributes set (and maybe Read Only, I don’t remember exactly now.). IBM PC DOS has similar files, IBMDOS.COM and IBMBIO.COM as well as COMMAND.COM.

**Comment by keirf** [comment, marked as answer]
Apart from having a bootable IMG file, your Gotek needs correct jumpering and configuration. For a PC/AT that would normally be S1 and JC. However your Gotek model (PCB says SFRKC30.AT2 I guess) does not have a jumper position JC! So you need a plain-text file on your USB stick called FF.CFG and containing just the line
```
interface = ibmpc
```
I would also suggest putting jumpers on both S0 and S1. With only one drive in the system it will do no harm and then onme or the other is certainly correct!

Seeing `0-` on the display is promising as it means FlashFloppy did at least detect and mount an IMG file.

**Comment by CWEastwood** [comment]
Hello thank you for the replies. I can confirm that the PCB is labeled SFRKC30.AT2. Great guess XD. Therefore, and at your advice, I only have jumpers on S1 and S0. 

Similarly, I have modified the ff.cfg to only include the line "interface = ibmpc".

Assuming all is good to go with that, it will be a matter of creating a proper image file. Thank you Hharte for the advice, I will see about using a bootable ms-dos image. 

You've all been very helpful. If anyone else has anything to add please do so.

**Comment by CWEastwood** [comment]
Hello again, 

I just wanted to make a final post for posterity's sake. I loaded the usb drive with 4 different unmodified dos boot images. I plugged in my storage device and cd rom (which up to this point I had left unplugged). Once powered on the 7 segment display did a little dance and the computer gave me the following screens

       "Award BootBlock BIOS v1.0
        Copyright (c) 1998, Award Software, Inc.

        BIOS ROM checksum error
        Keyboard error or no keyboard present ('cause I didn't have the keyboard plugged in)

        Detecting floppy drive A media

        Drive media is : 1.44MB
        Starting MS-DOS"

Followed by:

       "CD-ROM Device Driver for IDE (Four Channels Supported)
        (C)Copyright Oak Technology Inc. 1993-1996
        Driver version       : V340
        Device Name        : Banana
        No drives found, aborting installation

        Packed file is corruptA:\>"

So I'm not entirely sure what that means. Perhaps my BIOS was never corrupted to begin with and all it wanted was an OS. Either way, Gotek appears to be working as intended. 

Now I'm off to figure out what all this gobbledy-gook means.

Thank you all again!
CW Eastwood
